PowerValidatedSolutions.psm1

```diff
@@ -20714,6 +20721,384 @@ Function Get-PasswordPolicyConfig {
     $customConfig
 }
 
+Function checkRange() {
+	Param(
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$name,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$value,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$minRange,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [int]$maxRange,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [Bool]$required
+	)
+	if (($value -eq "Null") -and ($required -eq $true)) {
+		Write-LogMessage -Type ERROR -Message "$name parameter has not been configured."  -Colour Red
+		return $false
+	} elseif (($value -lt $minRange) -or ($value -gt $maxRange)) {
+		Write-LogMessage -Type ERROR -Message "The recommended range for $name should be between $minRange and $maxRange. [$value]"  -Colour Red
+		return $false
+	} else {
+		return $true
+	}
+}
+
+Function checkEmailString() {
+	Param(
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$name,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$address,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [Bool]$required
+	)
+	
+	if (($address -eq "Null") -and ($required -eq $true)) {
+		Write-LogMessage -Type ERROR -Message "$name variable has not been configured." -Colour Red
+		return $false
+	}
+	$checkStatement = $address -match "^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$"
+	if ($checkStatement -eq $true) {
+		return $true
+	} else {
+		Write-LogMessage -Type ERROR -Message "Please input a validate email address for $name " -Colour Red
+		return $false
+	}
+}
+
+Function Test-PasswordPolicyConfig {
+    Param (
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [psobject]$customConfig
+    )
+
+    # import default configuration JSON for comparing parameters
+    $defaultConfig = Get-PasswordPolicyDefault
+    $encounterError = "False"
+
+    # validating product Types in customConfig file
+    $defaultProductList = $defaultConfig | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+    $customProductList = $customConfig | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+    $defaultSection = "passwordExpiration", "passwordComplexity", "accountLockout"
+	
+    foreach ($product in $customProductList) {
+        if (-Not $defaultProductList.Name.Contains($product.Name)) {
+            Write-LogMessage -Type ERROR -Message "Unknow product type $($product.Name), please check the custom configuration file and run the cmdlet again" -Colour Red
```

Comment:
Change to:

Write-LogMessage -Type ERROR -Message "Found Unknown Product ($($product.Name)), Please Check the Password Policy Configuration File and Run Again" -Colour Red

Reply:
resolved in 2948bc273bae34312514db8aba5d7cb584efd8d4